### DISCUSSION TITLE
Add typings to extension result

### DISCUSSION THREAD

**Discussion by ScreamZ** [prisma/prisma#24440] [Q&A]
### Question

I've created such extension

```ts
import { Prisma } from "@prisma/client";

/**
 * An extension to get the proper content.
 */
export const ConversationMessageHelpersExtension = Prisma.defineExtension((prisma) =>
  prisma.$extends({
    name: "conversationMessageHelpers",
    result: {
      conversationMessage: {
        /**
         * Test JS DOC
         */
        contextualMessage: {
          needs: { content: true },
          /**
           * TTEST 2 JS DOC
           */
          compute({ content }) {
            return  typeof content === "string" ? content : content.text;
          },
        },
      },
    },
  }),
);

```

I wonder is there any way to add JSDoc to the created property, like we can with /// directive in prisma Schema ?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ScreamZ** [comment]
Currently i was just able to return a wrapper type, which is him typed and unlock issue
```ts
export const ConversationMessageHelpersExtension = Prisma.defineExtension((prisma) =>
  prisma.$extends({
    name: "conversationMessageHelpers",
    result: {
      conversationMessage: {
        helpers: {
          needs: { content: true },
          compute: ({ content }) => ({
            /**
             * Get the proper content of a message, whether it's a user message or an assistant message.
             */
            getContextualMessage: () => (typeof content === "string" ? content : content.text),
          }),
        },
      },
    },
  }),
);
```

**Reply by ludralph** [reply]
Hi @ScreamZ 👋 

Can you please explain what you mean by `him typed`? Also, can you please confirm if the issue was resolved?

**Reply by ScreamZ** [reply]
Sorry, I wrote at the end of the day and I'm not native English.

What I meant is that if I create a nested object, it is typed and JS doc is active. (Because extracted: `compute` uses `ReturnType` and therefore is inferred for type and JS doc)

Whereas if I rely on the compute ReturnType, I get no JS doc.


```
 conversationMessage: {
		/**
         * JS DOC ignored
         * /
        helpers: {
          needs: { content: true },
          compute: ({ content }) => ({
            /**
             * Get the proper content of a message, whether it's a user message or an assistant message.
             */
            getContextualMessage: () => (typeof content === "string" ? content : content.text),
          }),
        },
      },
```
`helpers` is typed without JS doc as an object (and therefore `getContextualMessage` has JS DOC)

Now if you write

```
 conversationMessage: {
		/**
         * JS DOC ignored
         * /
        helpers: {
          needs: { content: true },
          compute: ({ content }) => (typeof content === "string" ? content : content.text)
        },
      },
```

`helpers` is just a property that return a string, no JS doc involved.

**Reply by ludralph** [reply]
Hi @ScreamZ 

Can you please open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for this?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
